charon/src/ast/llbc_ast_utils.rs

```diff
@@ -130,6 +139,49 @@ impl Statement {
         }
     }
 
+    pub fn len(&self) -> usize {
+        match &self.content {
+            RawStatement::Sequence(seq) => seq.iter().map(|st| st.len()).sum(),
+            RawStatement::Switch(switch) => {
+                1 + switch.iter_targets().map(|st| st.len()).sum::<usize>()
+            }
+            RawStatement::Loop(st) => 1 + st.len(),
+            RawStatement::Nop => 0,
+            _ => 1,
+        }
+    }
+
+    /// Computes whether this statement diverges, and if yes returns a value indicating how far it
+    /// diverges (used in `prettify_cfg` as a heuristic).
+    pub fn diverges(&self) -> Option<impl Ord> {
+        // We prefer a continue/break to an abort to a sequence. We prefer continue/breaks to the
+        // innermost loop. And between a break and a continue to the same loop, we prefer the
+        // continue.
+        match &self.content {
+            RawStatement::Continue(loop_id) => Some((0, *loop_id, 0)),
+            RawStatement::Break(loop_id) => Some((0, *loop_id, 1)),
+            RawStatement::Return => Some((2, 0, 0)),
+            RawStatement::Abort(_) => Some((3, 0, 0)),
+            RawStatement::Sequence(seq) => seq
+                .iter()
+                .rev()
+                .filter(|st| !st.content.is_nop())
+                .next()
+                .and_then(|st| st.diverges()),
+            RawStatement::Switch(switch) => {
+                let mut min_div = (4, 0, 0);
+                for st in switch.iter_targets() {
+                    match st.diverges() {
+                        None => return None,
+                        Some(div) => min_div = std::cmp::min(div, min_div),
+                    }
+                }
+                Some(min_div)
```

Comment:
I'm not sure that I agree with this case.
If we have:
```
if b {
  st
} else {
  switch {
     ...
  }
}
```
and all the branches in the `switch` break the control-flow, do we really want to move `st`?
(I don't understand when this case happens)

Reply:
Same as above, happens if all the branches end with `break` or `continue`, e.g. [here](https://github.com/AeneasVerif/charon/pull/346/files#diff-8473fe4a377959b679ef87dddfe430b803ee027fe69bdc77546f20fcf6de9fbdL302-L338)

Reply:
The link doesn't point to something valid it seems

Reply:
Ah yeah, because I changed the code since I guess. We'll discuss again when I revive that change